File: app/src/main/java/com/stevenfrew/beatprompter/MIDIEvent.java
package com.stevenfrew.beatprompter;

import java.util.ArrayList;

class MIDIEvent extends BaseEvent {
    ArrayList<MIDIOutgoingMessage> mMessages;
    MIDIEventOffset mOffset;

    MIDIEvent(long time,ArrayList<MIDIOutgoingMessage> messages)
    {
        super(time);
        mMessages=messages;
    }

    MIDIEvent(long time,ArrayList<MIDIOutgoingMessage> messages,MIDIEventOffset offset)
    {
        this(time,messages);
        mOffset=offset;
    }
}

File: app/src/main/kotlin/com/stevenfrew/beatprompter/song/event/MidiEvent.kt
package com.stevenfrew.beatprompter.song.event

import com.stevenfrew.beatprompter.comm.midi.message.MidiMessage
import com.stevenfrew.beatprompter.midi.EventOffset

/**
 * A MIDIEvent tells the event processor to chuck some MIDI data out of the USB port.
 */
class MidiEvent(
	time: Long,
	val messages: List<MidiMessage>,
	val offset: EventOffset = EventOffset(0)
) : BaseEvent(time) {
	override fun offset(nanoseconds: Long): BaseEvent =
		MidiEvent(eventTime + nanoseconds, messages, offset)
}